modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/publisher/webapp/lifecycle/management/ChangeStateFromPublishToUnpublishTestCase.java

```diff
@@ -0,0 +1,155 @@
+        assertEquals(responseData.getString(AppmTestConstants.STATUS), "Success", "Changing WebApp life cycle state " +
+                "from publish to unpublish failed for user : " + userName + " who has sufficient privileges to change" +
+                " life cycle status.");
+    }
+
+    @Test(dataProvider = "inValidUserModeDataProvider", description = TEST_DESCRIPTION)
+    public void testChangeStateFromPublishToUnpublishWithInValidUsers(String userName, String password, String uuid)
+            throws Exception {
+        APPMPublisherRestClient publisherRestClient = new APPMPublisherRestClient(backEndUrl);
+        // Login to publisher by invalid user.
+        publisherRestClient.login(userName, password);
+        HttpResponse httpResponse = publisherRestClient.changeState(uuid, AppmTestConstants.LifeCycleStatus.UNPUBLISH);
+        JSONObject responseData = new JSONObject(httpResponse.getData());
+        // Logout from publisher by invalid user.
+        publisherRestClient.logout();
+        int responseCode = httpResponse.getResponseCode();
+        assertTrue(responseCode == 401, "Excepted status code is 401 for user :" + userName + ". But received " +
+                "status code is " + responseCode);
+        assertEquals(responseData.getString(AppmTestConstants.STATUS), "Access Denied", "Changing WebApp life " +
+                "cycle state from publish to unpublish allowed for user : " + userName + " who has insufficient " +
+                "privileges to change life cycle status.");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void closeDown() throws Exception {
+        // Deleted created WebApps by admin.
+        appmPublisherRestClient.deleteApp(app1Uuid);
+        appmPublisherRestClient.deleteApp(app2Uuid);
+        appmPublisherRestClient.deleteApp(app3Uuid);
+        // Logout from publisher by admin.
+        appmPublisherRestClient.logout();
+    }
+
+    private String createWebAppAndPublish(String appPrefix) throws Exception {
+        PolicyGroup defaultPolicyGroup = WebAppUtil.createDefaultPolicy();
+        HttpResponse response = appmPublisherRestClient.addPolicyGroup(defaultPolicyGroup);
+        String policyId = WebAppUtil.getPolicyId(response);
+        List<WebAppResource> webAppResources = WebAppUtil.createDefaultResources(policyId);
+        WebApp webApp = WebAppUtil.createBasicWebApp(adminUserName, appName + appPrefix, context + appPrefix,
+                                                     appVersion, "http://wso2.com/", webAppResources);
+        appmPublisherRestClient.createWebApp(webApp);
+        String appId = webApp.getAppId();
+        appmPublisherRestClient.publishWebApp(appId);
+        return appId;
+    }
+
+    @DataProvider
+    public Object[][] validUserModeDataProvider() throws Exception {
+        User adminUser = appMServer.getSuperTenant().getTenantAdmin();
+        User appPublisher = appMServer.getSuperTenant().getTenantUser("AppPublisher");
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/pull/357/commits/7796ffc89c20db3795ac7ed7bf4ed3bad835c843